src/dappsExplorer/useDappFavoritedToast.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { SectionList } from 'react-native'
+import ToastWithCTA from 'src/components/ToastWithCTA'
+import { Dapp } from 'src/dapps/types'
+
+const TOAST_DISMISS_TIMEOUT = 5000
+
+const useDappFavoritedToast = (sectionListRef: React.RefObject<SectionList>) => {
+  const { t } = useTranslation()
+
+  // do not use favoritedDapp to show / hide the toast, as the content of the toast depends on the dapp name during transitions
+  const [favoritedDapp, setFavoritedDapp] = useState<Dapp | null>(null)
+  const [showToast, setShowToast] = useState(false)
+
+  const timeoutIdRef = useRef<NodeJS.Timeout | null>(null)
```

Comment:
this bit is a bit convoluted, happy for any simplification suggestions. what i really want is to reset the timeout after every dapp favorite action, so that the toast stays on the screen for the full 5 seconds. before i started tracking the timeout id, favoriting a dapp then favoriting another dapp 4 seconds later = the first timeout got in the way and dismissed the toast for the second dapp 1 second later.

Reply:
I'm proposing a simplification, let me know if I missed a case: https://github.com/valora-inc/wallet/pull/3273

Reply:
thank you for this! 🧙🏻‍♂️